system-test/storage.ts

```diff
@@ -3022,17 +3021,25 @@ describe('storage', () => {
     });
 
     describe('second service account', () => {
-      before(async function () {
+      let accessId: string;
+
+      before(function () {
         if (!SECOND_SERVICE_ACCOUNT) {
           this.skip();
         }
-        await deleteHmacKeys(SECOND_SERVICE_ACCOUNT!, HMAC_PROJECT!);
+      });
+
+      after(async () => {
+        const hmacKey = storage.hmacKey(accessId, {projectId: HMAC_PROJECT});
+        await hmacKey.setMetadata({state: 'INACTIVE'});
+        await hmacKey.delete();
       });
 
       it('should create key for a second service account', async () => {
-        await storage.createHmacKey(SECOND_SERVICE_ACCOUNT!, {
+        const [hmacKey] = await storage.createHmacKey(SECOND_SERVICE_ACCOUNT!, {
```

Comment:
Does this have the same issue where these could be un-deleted if there's a test failure / system crash / operation fails? Do we need to loop over both service accounts in the global before hook?

Reply:
`Service Account` key will delete in delete test, while `Second Service Account` key will delete in `after` hook.
In case if any of above fails the `deleteStaleHmacKeys` will delete it in next cycle after one hour.

Reply:
Sorry, I missed that line of code 🤦🏻 Thanks!